data/provider/src/main/java/tech/pegasys/teku/api/ValidatorDataProvider.java

```diff
@@ -213,23 +217,19 @@ public SignedBeaconBlock parseBlindedBlock(
       final SignedBeaconBlock signedBeaconBlock) {
     return validatorApiChannel
         .sendSignedBlock(signedBeaconBlock.asInternalSignedBeaconBlock(spec))
-        .thenApply(
-            result -> {
-              int responseCode;
-              Optional<Bytes32> hashRoot = result.getBlockRoot();
-              if (result.getRejectionReason().isEmpty()) {
-                responseCode = SC_OK;
-              } else if (result
-                  .getRejectionReason()
-                  .get()
-                  .equals(FailureReason.INTERNAL_ERROR.name())) {
-                responseCode = SC_INTERNAL_ERROR;
-              } else {
-                responseCode = SC_ACCEPTED;
-              }
+        .thenApply(ValidatorDataProvider::generateSubmitSignedBlockResponse);
+  }
 
-              return new ValidatorBlockResult(responseCode, result.getRejectionReason(), hashRoot);
-            });
+  public SafeFuture<ValidatorBlockResult> submitSignedBlindedBlock(
+      final tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock signedBeaconBlock) {
+    LOG.debug("parsed block is from slot: {}", signedBeaconBlock.getMessage().getSlot());
+    final SpecMilestone milestone = spec.atSlot(signedBeaconBlock.getSlot()).getMilestone();
+    if (milestone == SpecMilestone.PHASE0 || milestone == SpecMilestone.ALTAIR) {
+      return validatorApiChannel
+          .sendSignedBlock(signedBeaconBlock)
+          .thenApply(ValidatorDataProvider::generateSubmitSignedBlockResponse);
+    }
+    return SafeFuture.failedFuture(new IllegalStateException("Not implemented"));
```

Comment:
This hard coding makes me nervous that we'll forget it and also that it will break our ability to produce bellatrix non-blinded blocks. I'd have thought we'd just pass through the block and let `ValidatorApiChannel` handle it.

Reply:
previously it'd bail well before now for all types, could throw a not implemented exception? my guess is the api channel will need to be a different interface?

Reply:
oh sorry I missed the blinded in the method name.  I though it was just `submitSignedBlock`.  I'd still just pass it through to `ValidatorApiChannel` - it can detect if the block is blinded or not and #5273 seems to be moving towards just handling that seamlessly.